### PR TITLE
Update sourcemap to rc-1.0

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

This PR updates all parcel packages to be compatible with the newest Rust version of the source-map library.

Made some breaking changes in the API as well, before it's too late (stable release)

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>React HackerNews 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>AtlasKit Editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>Three.js 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/298553055638454797)**

**Comment by DeMoorJasper** [comment]
`checkSourceMapping` is failing due to a change in the source-map library where we no longer match mappings after the requested index for `findClosestMapping`, this was to mimic the behaviour of `source-map` by Mozilla. This could easily be changed, not sure what we want here.

cc @mischnic

**Comment by mischnic** [comment]
You mean that `findClosestMapping` now returns the nearest mapping that comes before the passed position? I think that makes more sense anyway.

**Comment by DeMoorJasper** [comment]
@mischnic Yeah exactly, that's how other libraries work so changed it. In that case I'll update the test util to work with this change

File: packages/packagers/js/src/ScopeHoistingPackager.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ export class ScopeHoistingPackager {
 
       let [content, map, lines] = this.visitAsset(asset);
       if (sourceMap && map) {
-        sourceMap.addBufferMappings(map.toBuffer(), lineCount);
+        sourceMap.addBuffer(map.toBuffer(), lineCount);
```
Comment (devongovett) [inline packages/packagers/js/src/ScopeHoistingPackager.js]: Should we use `sourceMap.addSourceMap(map, lineCount)` here? I think it should improve performance a bit.

**Review by devongovett — CHANGES_REQUESTED** [review]

File: packages/packagers/js/src/ScopeHoistingPackager.js (hunk 1 above)
Comment (DeMoorJasper) [reply packages/packagers/js/src/ScopeHoistingPackager.js]: Oh yes definitely

**Comment by devongovett** [comment]
Looks like CI is failing on addSourceMap...

**Review by devongovett — APPROVED** [review]

**Pull request merged**